gtfu/src/web/graas-report-web.js

```diff
@@ -0,0 +1,343 @@
+        // but that occurs before image is loaded and scaleRatio is determined by image size
+        trips.forEach(function (item) {
+            item.map.x *= scaleRatio;
+            item.map.y *= scaleRatio;
+            item.map.width *= scaleRatio;
+            item.map.height *= scaleRatio;
+            item.timeline.x *= scaleRatio;
+            item.timeline.y *= scaleRatio;
+            item.timeline.width *= scaleRatio;
+            item.timeline.height *= scaleRatio;
+        });
+        drawReport();
+    };
+
+    window.addEventListener('resize', function(event) {
+
+        load();
+    });
+
+    document.body.addEventListener('click', function(event) {
+
+        drawReport();
+
+        scrollTop = document.documentElement.scrollTop; // Top of current screen
+        timelineScrollTop = scrollTop + dropdownHeight; // Top of current timeline
+        mapScrollTop = scrollTop + headerHeight;        // Top of current map
+        scrollBottom = scrollTop + windowHeight;        // Bottom of current map
+
+        for (var i = 0; i < trips.length; i++){
+            var searchY = event.pageY;
+            var searchObject;
+
+            // If the click occurs below the header, adjust y value and search map objects
+            if(searchY >= mapScrollTop){
+                searchY -=headerHeight;
+                searchObject = trips[i].map;
+
+            // If the click occurs within the header, adjust y value and search timeline objects
+            } else {
+                searchY -=timelineScrollTop;
+                searchObject = trips[i].timeline;
+            }
+            if (objectContainsPoint(searchObject, event.pageX, searchY)){
+                drawMetadata(trips[i]);
+                selectTrip(mapCtx, trips[i].map);
+                selectTrip(timelineCtx, trips[i].timeline);
+                scrollToTrip(trips[i])
+                break;
+            }
+        }
+    });
+}
+
+function objectContainsPoint(object, x, y){
+    if(x >= object.x){
+        if(x <= (object.x + object.width)){
+            if(y >= object.y ){
+                if(y <= (object.y + object.height)){
+                    return true;
+                }
+            }
+        }
+    }
+}
+
+function drawReport(){
+    console.log("Drawing map...");
+    timelineCtx.drawImage(img, 0, 0, imageWidth, timelineHeight, 0, 0, canvasWidth, timelineCanvasHeight);
+    mapCtx.drawImage(img, 0, timelineHeight, imageWidth, mapHeight, 0, 0, canvasWidth, mapCanvasHeight);
+}
+
+function drawMetadata(trip){
+    console.log("drawTooltip()");
+
+    var margin = 5;
+
+    metadataWidth = trip.map.width;
+    metadataX = trip.map.x + metadataWidth;
+    metadataY = trip.map.y;
+    metadataHeight = trip.map.height;
+
+    while ((tooltipItems.length * (font_size + margin) >= metadataHeight)
+            || (getMaxTextWidth(trip) + margin * 2 >= metadataWidth))  {
+        console.log("reducing font size...");
+        font_size -= 5;
```

Comment:
What happens when the font size goes negative?

Reply:
It shouldn't but that's a good question. I added some logic to simply not display the metadata if font size gets down to zero. This would occur if:
- Window is really small (possibly on mobile)
- Text is really really long